plugins/dashboard/dashboard.js

```diff
@@ -0,0 +1,348 @@
+                      screens.critical[beat.apikey].push(beat);
+                      screens.count.critical++;
+                    }
+                  }
+
+                  // Add all beats to the all array.
+                  if (!screens.all.hasOwnProperty(beat.apikey)) {
+                    screens.all[beat.apikey] = [];
+                  }
+                  screens.all[beat.apikey].push(beat);
+                  screens.count.total++;
+                }
+                deferred.resolve(screens);
+              },
+              function (error) {
+                deferred.reject(error);
+              })
+            });
+          }
+        },
+        function (error) {
+          deferred.reject(error);
+        }
+      );
+    },
+    function (error) {
+      deferred.reject(error);
+    }
+  );
+
+  return deferred.promise;
+};
+
+/**
+ * Helper function to load a screen.
+ *
+ * @param apikey
+ *   API-key for the screen to load.
+ * @param name
+ *   The name linked to the api-key.
+ * @param screenId
+ *   The id of screen to load.
+ *
+ * @returns {*}
+ */
+Dashboard.prototype.loadScreen = function loadScreen(apikey, name, screenId) {
+  var self = this;
+  var deferred = self.Q.defer();
+
+  var screen = new self.screen(apikey, screenId);
+  screen.load().then(
+    function (screenObj) {
+      deferred.resolve({
+        apikey: apikey,
+        name: name,
+        id: screenObj.id,
+        title: screenObj.title,
+        heartbeat: self.moment.unix(screenObj.heartbeat).format('D. MMMM YYYY HH:mm:ss'),
+        expired: self.expired(screenObj.heartbeat)
+      })
+    },
+    function (error) {
+      self.logger.error(error.message);
+      deferred.reject(error.message);
+    }
+  );
+
+  return deferred.promise;
+};
+
+/**
+ * Check if a given timestamp has expired.
+ *
+ * @param timestamp
+ *   Unix-timestamp.
+ *
+ * @returns
+ *   TRUE if expired else FALSE.
+ */
+Dashboard.prototype.expired = function expired(timestamp) {
+  return timestamp < Math.round((new Date()).getTime() / 1000) - 900;
+};
+
+/**
+ * Load black list form disk.
+ *
+ * @returns {*}
+ *   Promise that either will resolve when the data is ready or reject with an
+ *   error.
+ */
+Dashboard.prototype.load = function load() {
+  "use strict";
+
+  var self = this;
+  var deferred = self.Q.defer();
+
+  self.jf.readFile(self.config.blacklist, function(error, list) {
+    if (error) {
+      deferred.reject(new Error(error));
+    }
+    else {
+      deferred.resolve(list);
+    }
+  });
+
+  return deferred.promise;
+};
+
+/**
+ * Save black list to disk.
+ *
+ * @param string[] list
+ *  The black list with apikeys and screen ids as objects.
+ *
+ * @returns {*}
+ *   Promise that either will resolve when the data is saved or reject with an
+ *   error.
+ */
+Dashboard.prototype.save = function save(list) {
+  "use strict";
+
+  var self = this;
+  var deferred = self.Q.defer();
+
+  // Check that api-keys have been load first.
```

Comment:
Are you checking this? Or Copy/Paste error?

Reply:
ups